tests/scripts/quiet/make

```diff
@@ -0,0 +1,44 @@
+#! /usr/bin/env bash
+#
+# Copyright The Mbed TLS Contributors
+# SPDX-License-Identifier: Apache-2.0 OR GPL-2.0-or-later
+#
+# This swallows the output of the wrapped tool, unless there is an error.
+# This helps reduce excess logging in the CI.
+
+# If you are debugging a build / CI issue, you can get complete unsilenced logs
+# by un-commenting the following line (or setting VERBOSE_LOGS in your environment):
+# VERBOSE_LOGS=1
+
+# don't silence invocations containing these arguments
+NO_SILENCE=" --version | test "
+
+TOOL=$(basename "$0")
+
+# Locate original tool
+ORIGINAL_TOOL=$(type -ap ${TOOL} | grep -v "$0" | head -n1 )
+
+if [[ " $@ " =~ $NO_SILENCE || -n "${VERBOSE_LOGS}" ]]; then
+    ${ORIGINAL_TOOL} "$@"
+    EXIT_STATUS=$?
+else
+    # Display the command being invoked - if it succeeds, this is all that will
+    # be displayed.
+    echo "${TOOL} $@"
```

Comment:
Displaying the `make` invocation is something I'd like to do in all cases. It would be very helpful to re-create complex builds, such as builds with drivers, with tweaked arguments (typically enabling debug). Since you've already set up a `make` wrapper, please:

* Display this regardless of the verbosity setting.
* Use an unambiguous way to display the command line: it's vital to distinguish spaces inside arguments and spaces that separate arguments. `printf '%q ' "$@"; printf '\n'` is a bit ugly but works. (At least with bash 4, I need to confirm if macOS's antique bash already supports `%q`.)